data/scripts.ts

```diff
@@ -96,6 +96,7 @@ export const BattleScripts: BattleScriptsData = {
 			pokemon.side.zMoveUsed = true;
 		}
 		const moveDidSomething = this.useMove(baseMove, pokemon, target, sourceEffect, zMove, maxMove);
+		this.lastSuccessfulMoveThisTurn = moveDidSomething ? this.activeMove && this.activeMove.id : null;
```

Comment:
Wouldn't `this.activeMove?.id` have worked here?

Reply:
`this.activeMove?.id` short circuits with a value of `undefined` if `activeMove` is nullish. Admittedly, this is my first contribution to PS in awhile, but I seemed to remember the convention being `T | null` for optional properties like this. I could go back and update the type to `T | null | undefined` and be able to use optional chaining here.

Reply:
Clearly that means it should be `(moveDidSomething && this.activeMove?.activeMove.id) || null`

Reply:
Eh, I personally wouldn't lean towards that as I think that's less readable then just skipping the optional chaining here, but I can make that update if people agree on it.

Reply:
Oh, it does? That's annoying. Never mind then.

Reply:
I was half-shitposting. I agree that `&&` being used for its return value is less readable than `||` being used for its return value.